File: coding_agent/code/EntityServiceTest.java
package com.example.service;

import com.example.entity.Entity;
import com.example.repository.EntityRepository;
import org.springframework.stereotype.Service;
import java.util.List;
import java.util.Optional;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;
import static org.mockito.Mockito.*;
import static org.junit.jupiter.api.Assertions.*;

@ExtendWith(MockitoExtension.class)
public class EntityServiceTest {

    @Mock
    private EntityRepository entityRepository;

    @InjectMocks
    private EntityService entityService;

    @Test
    void getAllEntities_ShouldReturnListOfEntities() {
        var mockEntities = List.of(new Entity(1L, "Entity1"), new Entity(2L, "Entity2"));
        when(entityRepository.findAll()).thenReturn(mockEntities);

        var entities = entityService.getAllEntities();

        assertEquals(2, entities.size());
        assertEquals("Entity1", entities.get(0).getName());
    }
}

File: coding_agent/code/kotlin/EntityServiceTest.kt
package com.example.service

import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.mockito.InjectMocks
import org.mockito.Mock
import org.mockito.junit.jupiter.MockitoExtension
import org.mockito.kotlin.whenever
import org.mockito.kotlin.verify
import org.mockito.kotlin.times
import org.mockito.kotlin.mock
import kotlin.test.assertEquals

@ExtendWith(MockitoExtension::class)
class EntityServiceTest {

    @Mock
    lateinit var entityRepository: EntityRepository

    @InjectMocks
    lateinit var entityService: EntityService

    @Test
    fun `getAllEntities should return list`() {
        val mockEntities = listOf(Entity(1L, "Entity1"), Entity(2L, "Entity2"))
        whenever(entityRepository.findAll(org.mockito.kotlin.any())).thenReturn(org.springframework.data.domain.PageImpl(mockEntities))

        val entities = entityService.getAllEntities(0, 10)
        assertEquals(2, entities.content.size)
    }
}
